test/jdk/java/awt/Frame/InvisibleOwner/InvisibleOwner.java

```diff
@@ -24,99 +24,158 @@
-        if (c == null) {
-            throw new RuntimeException("Robot.getPixelColor() failed");
-        }
-        if (c.equals(frame.getBackground())) {
-            throw new RuntimeException("The invisible frame has become visible");
-        }
-        if (!c.equals(helperFrame.getBackground())) {
-            throw new RuntimeException("The background helper frame has been covered by something unexpected");
-        }
+            Rectangle helperBounds = helperFrame.getBounds();
+            System.out.println("helperFrame bounds = " + helperBounds);
+            if (!helperBounds.contains(C_X, C_Y)) {
+                System.out.println("Helper not positioned where it needs to be");
+                return;
+            }
 
-        // 2. Try to click it
-        robot.mouseMove(F_X + F_W / 2, F_Y + F_H / 2);
-        robot.mousePress(InputEvent.BUTTON1_MASK);
-        robot.mouseRelease(InputEvent.BUTTON1_MASK);
-        try { Thread.sleep(500); } catch (Exception ex) {}
+            // Clicking the owned window shouldn't make its owner visible
+            Rectangle ownedWindowBounds = ownedWindow.getBounds();
+            robot.mouseMove(ownedWindowBounds.x + ownedWindowBounds.width / 2,
+                            ownedWindowBounds.y + ownedWindowBounds.height / 2);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            robot.delay(1000);
 
-        // Cleanup
-        window.dispose();
-        frame.dispose();
-        helperFrame.dispose();
+            // 1. Check the color at the center of the invisible & helper frame location
+            Color c = robot.getPixelColor(C_X, C_Y);
+            System.out.println("Sampled pixel at " + C_X +"," + C_Y);
+            System.out.println("Pixel color: " + c);
+            if (c == null) {
+                captureScreen();
+                throw new RuntimeException("Robot.getPixelColor() failed");
+            }
+            if (c.equals(invisibleFrameBgColor)) {
+                captureScreen();
+                throw new RuntimeException("The invisible frame has become visible");
+            }
+            if (!c.equals(helperFrameBgColor)) {
+                captureScreen();
+                throw new RuntimeException(
+                    "Background frame was covered by something unexpected");
+            }
 
-        // Final checks
-        if (invisibleOwnerClicked) {
-            throw new RuntimeException("An invisible owner frame got clicked. Looks like it became visible.");
-        }
-        if (!backgroundClicked) {
-            throw new RuntimeException("The background helper frame hasn't been clciked");
+            // 2. Try to click it - event should be delivered to the
+            // helper frame, not the invisible frame.
+            robot.mouseMove(C_X, C_Y);
+            robot.mousePress(InputEvent.BUTTON1_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_MASK);
```

Comment:
In other places,we have modified to use BUTTON1_DOWN_MASK, maybe we can do here too.